Content.Server/Botany/Systems/PlantAnalyzerSystem.cs

```diff
@@ -0,0 +1,261 @@
+            }
+        }
+        else if (plantcomp != null) // If we poke the plantholder, it checks the plantholder seed.
+        {
+            seedData = plantcomp.Seed;
+            if (seedData != null)
+            {
+                state = ObtainingGeneDataSeed(seedData, target, true, component.Comp.AdvancedScan);  // SeedData is a unique seed in a tray.
+            }
+        }
+
+        if (state == null)
+            return;
+
+        _uiSystem.SendUiMessage(ui, state);
+    }
+
+    /// <summary>
+    ///     Analysis of seed from prototype.
+    /// </summary>
+    public PlantAnalyzerScannedSeedPlantInformation ObtainingGeneDataSeedProt(SeedPrototype comp, EntityUid target, Boolean scanMode)
+    {
+        var seedName = "\n  " + Loc.GetString("plant-analyzer-window-label-trademark") +
+            "\n    " + Loc.GetString(comp.DisplayName);
+
+        var seedChem = "\n   ";
+        var plantHarvestType = "";
+        var exudeGases = "";
+
+        if (comp.HarvestRepeat == HarvestType.Repeat) plantHarvestType = Loc.GetString("plant-analyzer-harvest-repeat");
+        if (comp.HarvestRepeat == HarvestType.NoRepeat) plantHarvestType = Loc.GetString("plant-analyzer-harvest-ephemeral");
+        if (comp.HarvestRepeat == HarvestType.SelfHarvest) plantHarvestType = Loc.GetString("plant-analyzer-harvest-autoharvest");
+
+        seedChem += String.Join("\n   ", comp.Chemicals.Select(item => item.Key.ToString()));
+
+        exudeGases = comp.ExudeGasses.Count > 0
+            ? string.Join("\n   ", comp.ExudeGasses.Keys)
+            : Loc.GetString("plant-analyzer-plant-gasses-no");
+
+        List<string> tolerancesList = CheckAllTolerances(comp);
+        List<string> generalTraitsList = CheckGeneralTraits(comp);
+        List<string> mutationsList = CheckAllMutations(comp);
+
+        return new PlantAnalyzerScannedSeedPlantInformation(GetNetEntity(target), seedName,
+            seedChem, plantHarvestType, exudeGases, mutationsList, false, tolerancesList, generalTraitsList, scanMode);
+    }
+
+    /// <summary>
+    ///     Analysis of unique seed.
+    /// </summary>
+    public PlantAnalyzerScannedSeedPlantInformation ObtainingGeneDataSeed(SeedData comp, EntityUid target, bool trayChecker, Boolean scanMode)
+    {
+        var seedName = Loc.GetString(comp.DisplayName);
+
+        var seedChem = "\n   ";
+        var plantHarvestType = "";
+        var exudeGases = "";
+
+        if (comp.HarvestRepeat == HarvestType.Repeat) plantHarvestType = Loc.GetString("plant-analyzer-harvest-repeat");
+        if (comp.HarvestRepeat == HarvestType.NoRepeat) plantHarvestType = Loc.GetString("plant-analyzer-harvest-ephemeral");
+        if (comp.HarvestRepeat == HarvestType.SelfHarvest) plantHarvestType = Loc.GetString("plant-analyzer-harvest-autoharvest");
+
+        seedChem += string.Join("\n   ", comp.Chemicals.Select(item => item.Key.ToString()));
+
```

Comment:
Don't just copy-paste to get strings out, just convert it to the string, and also please format your code.